File: src/main/java/me/ste/ivremotecontrol/block/decorinterface/decor/TileEntityPoleWrapper.java
package me.ste.ivremotecontrol.block.decorinterface.decor;

import minecrafttransportsimulator.blocks.components.ABlockBase;
import minecrafttransportsimulator.blocks.tileentities.components.ATileEntityPole_Component;
import minecrafttransportsimulator.blocks.tileentities.instances.TileEntityPole;
import minecrafttransportsimulator.jsondefs.JSONText;
import minecrafttransportsimulator.packets.instances.PacketEntityTextChange;

import java.util.ArrayList;
import java.util.List;

public class TileEntityPoleWrapper implements Decor {
    private TileEntityPole pole;

    public TileEntityPoleWrapper(TileEntityPole pole) {
        this.pole = pole;
    }


    @Override
    public List<String> getTextLines(ABlockBase.Axis axis) {
        ATileEntityPole_Component component = this.pole.components.get(axis);
        if(component != null) {
            return new ArrayList<>(component.text.values());
        } else {
            return null;
        }
    }

    @Override
    public void setTextLines(ABlockBase.Axis axis, List<String> lines) {
        ATileEntityPole_Component component = this.pole.components.get(axis);
        if(component != null) {
            component.updateText(lines);
        }
    }

    @Override
    public PacketEntityTextChange getUpdatePacket(ABlockBase.Axis axis, List<String> lines) {
        ATileEntityPole_Component component = this.pole.components.get(axis);
        if(component != null) {
            return new PacketEntityTextChange(component, lines);
        } else {
            return null;
        }
    }

    @Override
    public List<JSONText> getDefinitionTextObjects(ABlockBase.Axis axis) {
        ATileEntityPole_Component component = this.pole.components.get(axis);
        if(component != null) {
            return component.definition.rendering.textObjects;
        } else {
            return null;
        }
    }

    @Override
    public boolean isPole() {
        return true;
    }
}

File: src/main/kotlin/me/ste/ivremotecontrol/block/decorinterface/decor/TileEntityPoleWrapper.kt
package me.ste.ivremotecontrol.block.decorinterface.decor

import minecrafttransportsimulator.blocks.components.ABlockBase
import minecrafttransportsimulator.blocks.tileentities.instances.TileEntityPole
import minecrafttransportsimulator.packets.instances.PacketEntityTextChange

class TileEntityPoleWrapper(private val pole: TileEntityPole) : Decor() {
    override fun getTextLines(axis: ABlockBase.Axis) = this.pole.components[axis]?.text
    override val isPole = true

    override fun setTextLines(axis: ABlockBase.Axis, lines: LinkedHashMap<String, String>) {
        this.pole.components[axis]?.updateText(lines)
    }

    override fun getUpdatePacket(axis: ABlockBase.Axis, lines: LinkedHashMap<String, String>) =
        this.pole.components[axis]?.let { PacketEntityTextChange(it, lines) }
}